addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1902,17 +1902,49 @@ var SnippetsMenu = Widget.extend({
             const $invisibleSnippets = globalSelector.all().find('.o_snippet_invisible').addBack('.o_snippet_invisible');
 
             $invisibleDOMPanelEl.toggleClass('d-none', !$invisibleSnippets.length);
-
-            const proms = _.map($invisibleSnippets, async el => {
-                const editor = await this._createSnippetEditor($(el));
+            // childrenPerSnippet -key: a snippet.
+            //                    -value: an array filled with its children
+            // snippets.
+            const childrenPerSnippet = new Map();
+            // Filter the $invisibleSnippets to only keep the non children
+            // snippets and create the map (childrenPerSnippet) of the snippets
+            // and their children.
+            const $rootInvisibleSnippets = $invisibleSnippets.filter((index, childSnippetEl) => {
+                for (const parentSnippetEl of $invisibleSnippets) {
+                    if (childSnippetEl !== parentSnippetEl
+                        && parentSnippetEl.contains(childSnippetEl)) {
+                        const childrenSnippets = childrenPerSnippet.get(parentSnippetEl) || [];
+                        childrenPerSnippet.set(parentSnippetEl, [...childrenSnippets, childSnippetEl]);
+                        return false;
+                    }
+                }
+                return true;
+            });
+            // Insert an invisibleSnippet in the DOM.
+            const createInvisibleElement = async (invisibleSnippetEl, isChild = false) => {
+                const editor = await this._createSnippetEditor($(invisibleSnippetEl));
```

Comment:
Does it make sense to `_createSnippetEditor` if the snippet is not visible ? :thinking:

Reply:
In fact, it was already the case before this commit. One advantage of doing so is to use the existing `editor.getName()` and `editor.isTargetVisible()`. But maybe it is not worth creating an editor only to use those methods and maybe it is better to redefined those one to be able to use them without creating an editor?

Reply:
Indeed it was done before so I suppose it's ok to keep this :+1: